src/Simplex/Chat.hs

```diff
@@ -651,9 +651,28 @@ processChatCommand = \case
         ChatErrorAgent (SMP SMP.AUTH) -> pure $ CRRcvFileAcceptedSndCancelled ft
         ChatErrorAgent (CONN DUPLICATE) -> pure $ CRRcvFileAcceptedSndCancelled ft
         e -> throwError e
-  CancelFile fileId -> withUser $ \user@User {userId} -> do
-    ft <- withStore (\st -> getFileTransfer st userId fileId)
-    withChatLock . procCmd $ cancelFile user fileId ft
+  CancelFile fileId -> withUser $ \user@User {userId} ->
+    withChatLock . procCmd $ do
+      alreadyCancelled <- cancelFile user fileId
+      ft <- withStore $ \st -> getFileTransfer st userId fileId
+      case ft of
+        FTSnd _ _ ->
+          unless alreadyCancelled $ do
+            sharedMsgId <- withStore $ \st -> getSharedMsgIdByFileId st userId fileId
+            void $
+              withStore (\st -> getContactOrGroupIdByFileId st userId fileId) >>= \case
+                Left contactId -> do
```

Comment:
Probably ChatRef type is better for it, Left/Right feels a hack

Reply:
There are many contexts where only Direct/Group is allowed, so we could narrow it down with typefamily or just show internal error